docs/API.md

```diff
@@ -104,132 +105,161 @@ gulp.task('somename', function() {
+You should either provide the task name or avoid anonymous functions.
 
 #### fn
 
-The function that performs the task's operations. Generally this takes the form of `gulp.src().pipe(someplugin())`.
+The function that performs the task's operations. 
+Generally this takes the form of `gulp.src().pipe(someplugin())`.
 
 #### Async task support
 
 Tasks can be made asynchronous if its `fn` does one of the following:
 
-##### Accept a callback
-
-```javascript
-// run a command in a shell
-var exec = require('child_process').exec;
-gulp.task('jekyll', function(cb) {
-  // build Jekyll
-  exec('jekyll build', function(err) {
-    if (err) return cb(err); // return error
-    cb(); // finished task
-  });
-});
-```
-
 ##### Return a stream
 
 ```js
 gulp.task('somename', function() {
-  var stream = gulp.src('client/**/*.js')
+  return gulp.src('client/**/*.js')
     .pipe(minify())
     .pipe(gulp.dest('build'));
-  return stream;
 });
 ```
 
-##### Return a promise
+##### Accept a callback
 
-```javascript
-var Q = require('q');
+```js
+var del = require('del');
 
-gulp.task('somename', function() {
-  var deferred = Q.defer();
+gulp.task('clean', function(done) {
+  del(['.build/'], function(err) {
+    done(err)
+  });
+  // or
+  // del(['.build/'], done);
+});
+```
 
-  // do async stuff
-  setTimeout(function() {
-    deferred.resolve();
-  }, 1);
+##### Return a promise
 
-  return deferred.promise;
+```js
+var Promise = require('promise');
+var del = require('del');
+
+gulp.task('clean', function() {
+  return new Promise(function (resolve, reject) {
+    del(['.build/'], function(err) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve();
+      }
+    });
+  });
 });
 ```
 
-**Note:** By default, tasks run with maximum concurrency -- e.g. it launches all the tasks at once and waits for nothing. If you want to create a series where tasks run in a particular order, you need to do two things:
+or:
+```js
+var promisedDel = require('promised-del');
 
-- give it a hint to tell it when the task is done,
-- and give it a hint that a task depends on completion of another.
+gulp.task('clean', function() {
+  return promisedDel(['.build/']);
+});
+```
 
-For these examples, let's presume you have two tasks, "one" and "two" that you specifically want to run in this order:
+##### Return a child process
 
-1. In task "one" you add a hint to tell it when the task is done.  Either take in a callback and call it when you're
-done or return a promise or stream that the engine should wait to resolve or end respectively.
+```js
+gulp.task('clean', function() {
+  return spawn('rm', ['-rf', path.join(__dirname, 'build')]);
+});
 
-2. In task "two" you add a hint telling the engine that it depends on completion of the first task.
+```
 
-So this example would look like this:
 
-```js
-var gulp = require('gulp');
+### gulp.parallel(tasks, ...)
+
+Takes a variable amount of task names or function and returns
```

Comment:
need to mention that if you use task names, tasks must be registered prior to calling gulp.parallel (same goes for series)